src/api/events.ts

```diff
@@ -1,9 +1,44 @@
 import { Event } from '../types/domain';
 
+let EVENTS_CACHE: Array<Event> | null = null;
+const EVENT_CACHE: Record<string, Event> = {};
```

Comment:
If we need caching, could the cache functionality be wrapped into a class/component of some kind so that the API and lifecycle of thing is a little more clear?

I think one issue I see here is that the events cache is never invalidated, because I see `EVENTS_CACHE` is set during `cacheEvents`, but never reset to `null` again, so the `fetchEvents` promise will always be resolved with the cache after the initial load? Or am I misunderstanding the lifecycle here? I see the same issue for `EVENT_CACHE`. I think if I were to implement this myself I would try to make some client that wrapped the fetch API so that we could have some middleware where the caching concerns are handled (the middleware could activate based on url path for example, making sure it handles all CRUD operations on a resource since the API is more or less REST-like). As it is, caching code is kind of leaking out into several places which makes things a little hard to read.

Also I'm guessing the reason you have caching here is because it speeds up the transition from events view to the event view? I'd probably be a little sceptical on the latency gains here but I'm all for it for learning purposes, because caching can be really tricky 😃

Reply:
Yeah, I agree on all points. The actual impact of this caching implementation is minimal. I think the browser's own native caching for GET resources is better (given that the backend support this).

You are correct in reading that `EVENT_CACHE` is never invalidated. If we do go for the HTTP client approach then we could add a timer that regularly invalidates cache entries. I did initially start on such an implementation, but I felt that it quickly became too complex for this PR, so I opted for this simpler approach instead.

I can remove the cache stuff from this PR and leave it for another independent one.